base_tier_validation_waiting/models/tier_validation.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class TierValidation(models.AbstractModel):
+    _inherit = "tier.validation"
+
+    def _notify_review_available(self, tier_reviews):
+        """method to notify when reaching pending"""
+        subscribe = "message_subscribe"
+        post = "message_post"
+        if hasattr(self, post) and hasattr(self, subscribe):
+            for rec in self:
+                users_to_notify = tier_reviews.filtered(
+                    lambda r: r.definition_id.notify_on_pending and r.res_id == rec.id
+                ).mapped("reviewer_ids")
+                # Subscribe reviewers and notify
+                getattr(rec, subscribe)(
+                    partner_ids=users_to_notify.mapped("partner_id").ids
+                )
+                getattr(rec, post)(
+                    subtype_xmlid=self._get_requested_notification_subtype(),
+                    body=rec._notify_requested_review_body(),
+                )
+
+    def _validate_tier(self, tiers=False):
+        """extend base tier validation method to allow bypass of waiting
+        sequences as pending state is hardcoded"""
+        self.ensure_one()
+        super(TierValidation, self)._validate_tier(tiers)
+        tier_reviews = tiers or self.review_ids
+        if tier_reviews.filtered(
+            lambda r: r.status == "waiting"
+            and (self.env.user in r.reviewer_ids)
+            and r.approve_sequence_bypass
+        ):
+            tier_reviews.write(
```

Comment:
This should only update the filtered reviews, as otherwise all reviews, regardless of state and reviewer will be updated.